scripts/plot_network.py

```diff
@@ -182,7 +181,7 @@ def plot_map(n, ax=None, attribute='p_nom', opts={}):
     return fig
 
 
-def plot_total_energy_pie(n, ax=None):
+def plot_total_energy_pie(n, ax=None, opts={}):
```

Comment:
if opts is empty, then the function will throw an `KeyError`. Would suggest, therefore, that it's not an optional kwarg.

```suggestion
def plot_total_energy_pie(n, opts, ax=None):
```

Reply:
> thanks @thesethtruth for taking care! Just some minor comments, what do you think?

Hi Martha! Thanks for your comments. Sorry for the delayed response, I had a holiday break. I will start working on your comments right away.

Reply:
Completely agree with you on the non-optionality of the argument. I implemented the same scheme as I saw in the existing code, as not to break any implementations further down the line. Let me go over those arguments and check whether I can indeed change those as you suggest.